analyses/chromosomal-instability/util/chr-break-plot.R

```diff
@@ -0,0 +1,118 @@
+# Functions for chromosomal instability plots
+#
+# C. Savonen for ALSF - CCDL
+#
+# 2020
+
+map_density_plot <- function(granges,
+                             y_val,
+                             y_lab,
+                             color,
+                             main_title) {
+  # Given a GRanges object, plot it y value along its chromosomal mappings using
+  # ggbio.
+  #
+  # Args:
+  #   granges: A Granges object to plot
+  #   y_val: a character string of the columnname in listData spot of the
+  #          GenomicRanges to plot on the y axis
+  #   color: a color parameter
+  #   y_lab: a character string to use for the ylabel. Will be passed to
+  #          ggplot2::ylab argument.
+  #   main_title: a character string to use for the main title.
+  #
+  # Returns:
+  #  ggplot of chromosomal mapping of the y value given.
+  #
+  # For setting the scale later, need to get y's max
+  max_y <- max(
+    data.frame(granges@elementMetadata@listData) %>%
+      dplyr::pull(
+      !!rlang::sym(y_val)
+    ),
+    na.rm = TRUE
+  )
+  # Make the density plot
+  density_plot <- ggbio::autoplot(granges, ggplot2::aes(y = !!rlang::sym(y_val)),
```

Comment:
Just FYI, there is another strategy for supplying args specifically to ggplot: `aes_string()` rather than `aes()`. You don't need it here, but for future reference it's an easier way to pass in columns if all you need is `aes()`'ing them. Facets....who can say....

Reply:
Good to know. Thanks for that tip!